### PR TITLE
[Merged by Bors] - feat: properties of rnDeriv

### PR DESCRIPTION
**Pull request by RemyDegenne**
Various results about `rnDeriv`, notably `rnDeriv_add`, `rnDeriv_smul_left` and `rnDeriv_smul_right`. These results describe the Radon-Nikodym derivatives of sums and scaling of measures.

These lemmas were already there for signed measures, but not for `Measure`. The proofs for signed measures use that addition is cancelative (`μ + ν₁ = μ + ν₂ ↔ ν₁ = ν₂`). This is not true in general for measures, but is true when `μ` is mutually singular with the two other measures or when `μ` is sigma-finite, which is enough for these proofs.

---

- [x] depends on: #7544
- [x] depends on: #7714
- [x] depends on: #8032
- [x] depends on: #8026
- [x] depends on: #8040

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

File: Mathlib/MeasureTheory/Decomposition/Lebesgue.lean [hunk 1]
```diff
@@ -153,6 +179,19 @@ theorem withDensity_rnDeriv_le (μ ν : Measure α) : ν.withDensity (μ.rnDeriv
...
+  have h_dec := haveLebesgueDecomposition_add ν μ
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · rw [h, add_zero] at h_dec
+    rw [h_dec]
+    exact (mutuallySingular_singularPart ν μ).symm
+  · rw [h_dec, MutuallySingular.add_right_iff] at h
+    rw [← MutuallySingular.self_iff]
+    refine MutuallySingular.mono_ac h.2 ?_ AbsolutelyContinuous.rfl
+    exact withDensity_absolutelyContinuous _ _
+
```
Comment (JasonKYi) [inline Mathlib/MeasureTheory/Decomposition/Lebesgue.lean:194]: Can you add this too?
```suggestion
lemma rnDeriv_eq_zero (μ ν : Measure α) [ν.HaveLebesgueDecomposition μ] :
    ν.rnDeriv μ =ᵐ[μ] 0 ↔ μ ⟂ₘ ν := by
  rw [← withDensity_rnDeriv_eq_zero]
  refine ⟨fun h => ?_, withDensity_eq_zero <| (measurable_rnDeriv _ _).aemeasurable⟩
  sorry -- Follows from #8045
```

Comment (RemyDegenne) [reply Mathlib/MeasureTheory/Decomposition/Lebesgue.lean:197]: Done, thanks!

**Comment by JasonKYi** [comment]
Looks good, thanks!

maintainer merge

**Pull request closed without merging**